frontend/src/components/charts/custom-edges/straight-edge.tsx

```diff
@@ -1,8 +1,10 @@
 import { BaseEdge, type EdgeProps, getStraightPath } from '@xyflow/react';
 
-export function StraightEdge({ id, sourceX, sourceY, targetX, targetY, style }: EdgeProps) {
+export function StraightEdge({ id, sourceX, sourceY, targetX, targetY, style, data }: EdgeProps) {
     // Define a margin to stop the edge before the node boundary (hard coded... based on size of node would be better code practice)
     const margin = 35;
+    // How to write this better?
+    const dimmed = data?.dimmed as boolean | undefined;
```

Comment:
I assume you want to default to `false` when undefined. So you could write it as follows:

```suggestion
    const dimmed: boolean = data?.dimmed ?? false;
```

(Explicitly writing the type as boolean is not needed I believe, but your `as boolean | undefined` also should not be).

Reply:
the false is indeed better, the casting is needed because data?.dimmed is unknown and cannot be assigned to a boolean